### PR TITLE
Feature/add aatf new org

### PR DESCRIPTION
**Pull request by willdot**
PBI: 70849

Add a new organisation and then continue with that org to add an AATF to it.

Added auth to the AddAttf controller so that only internal admins can access it, even if they try to use a url instead of the button which is now hidden on the manage aatfs screen unless the user is an internal admin,

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs [hunk 1]
```diff
@@ -111,11 +120,139 @@ public async Task<ActionResult> Add(AddAatfViewModel viewModel)
...
+            {
+                IList<CountryData> countries = await GetCountries();
+
+                model.Address.Countries = countries;
+                return View(model);
+            }
+
+            using (var client = apiClient())
+            {
+                CreateOrganisationAdmin request = new CreateOrganisationAdmin()
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Add a unit test for this... when model is valid a request is created with valid parameters

Can we also add a similar unit test for the Add AATF post method so the AddAtf request is called with correct parameters

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+    using EA.Weee.Web.Constant;
+    using EA.Weee.Web.Filters;
     using EA.Weee.Web.Infrastructure;
     using System;
     using System.Collections.Generic;
     using System.Linq;
     using System.Threading.Tasks;
     using System.Web.Mvc;
 
+    [AuthorizeInternalClaimsAttribute(Claims.InternalAdmin)]
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Add unit test to check this attribute is on the controller

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs [hunk 3]
```diff
@@ -52,11 +55,18 @@ public async Task<ActionResult> Details(Guid id)
...
-        
+
+        [HttpGet]
         public async Task<ActionResult> ManageAatfs()
         {
             SetBreadcrumb();
-            return View(new ManageAatfsViewModel { AatfDataList = await GetAatfs() });
+
+            using (var client = apiClient())
+            {
```
[the remark this answers is not in the document]
Comment (willdot) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Done with tests

File: src/EA.Weee.Requests/Users/CheckUserRole.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+{
+    using EA.Prsd.Core.Mediator;
+    using EA.Weee.Security;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    public class CheckUserRole : IRequest<bool>
```
[the remark this answers is not in the document]
Comment (willdot) [reply src/EA.Weee.Requests/Users/CheckUserRole.cs]: Removed

File: src/EA.Weee.RequestHandlers/Admin/CreateOrganisationAdminHandler.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        private readonly WeeeContext db;
+
+        public CreateOrganisationAdminHandler(IWeeeAuthorization authorization, IGenericDataAccess dataAccess, WeeeContext context)
+        {
+            this.authorization = authorization;
+            this.dataAccess = dataAccess;
+            this.db = context;
+        }
+
+        public async Task<Guid> HandleAsync(CreateOrganisationAdmin message)
```
[the remark this answers is not in the document]
Comment (willdot) [reply src/EA.Weee.RequestHandlers/Admin/CreateOrganisationAdminHandler.cs]: There are some unit tests already :)

File: src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs [hunk 6]
```diff
@@ -41,6 +42,15 @@ public async Task<bool> HandleAsync(AddAatf message)
 
             Operator @operator = await dataAccess.GetSingleByExpression<Operator>(new OperatorByOrganisationIdSpecification(message.OrganisationId));
 
+            if (@operator == null)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs]: Can we add a unit test to check if the operator is null a operator is added to the context and the aatf used this operator

File: src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs [hunk 7]
```diff
@@ -41,6 +42,15 @@ public async Task<bool> HandleAsync(AddAatf message)
...
+                Organisation organisation = await dataAccess.GetById<Organisation>(message.OrganisationId);
+
+                @operator = new Operator(organisation);
+
+                await dataAccess.Add<Operator>(@operator);
+            }
+
             AatfContact contact = contactMapper.Map(message.AatfContact);
 
             Aatf aatf = new Aatf(
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs]: In the unit test can we check that the AATF is added to the context with all of the matching arguments... at the moment its currently only name

Comment (willdot) [reply src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs]: Done

File: src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs (hunk 6 above)
Comment (willdot) [reply src/EA.Weee.RequestHandlers/Admin/AddAatfRequestHandler.cs]: Done

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs (hunk 2 above)
Comment (willdot) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Added but can't test if it has the Claims.InternalAdmin because it's a private member

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs (hunk 1 above)
Comment (willdot) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Done
